charts/posthog/templates/secrets.yaml

```diff
@@ -16,6 +16,16 @@ data:
   {{- else }}
   posthog-secret: {{ randAlphaNum 40 | b64enc | quote -}}
   {{- end }}
+
+  {{- $clickhousePrevPassword := index ($previous.data | default dict) "clickhouse-password" }}
```

Comment:
Code smell: the index call here different than on line 43 due to that checking if `$previous` exists first and here using a default. Consistency would be :kissing:

Reply:
Here I want it to be `nil` if we are doing an install (i.e. previous didn't exist) or if the "clickhouse-password" didn't exist (i.e. first upgrade after this change - to move folks away from using the hardcoded "a1..." if they haven't specified a password themselves). I could use an if/else block, but that seems worse. 
```
{{- if $previous }}
{{- $clickhousePrevPassword := index $previous.data "clickhouse-password" }}
{{- else }}
{{- $clickhousePrevPassword := nil }}
```

Similarly below for uniqName it's easier to use the previous check earlier as that applies to the whole block, I only ever want to fail there if we're upgrading (vs here I want to generate & change the password if previously we were using the hardcoded a1...). But if I were to try to get rid of the inconsistency I'd change it there, that however didn't seem to belong to this pr.